lib/FrontendTool/FrontendTool.cpp

```diff
@@ -948,9 +948,13 @@ static bool performCompile(CompilerInstance &Instance,
 
   // We've just been told to perform a typecheck, so we can return now.
   if (Action == FrontendOptions::ActionType::Typecheck) {
-    const bool hadPrintAsObjCError = printAsObjCIfNeeded(
-        Invocation.getObjCHeaderOutputPathForAtMostOnePrimary(),
-        Instance.getMainModule(), opts.ImplicitObjCHeaderPath, moduleIsPublic);
+    const bool hadPrintAsObjCError =
+        Invocation.getFrontendOptions()
+            .InputsAndOutputs.hasObjCHeaderOutputPath() &&
+        printAsObjCIfNeeded(
+            Invocation.getObjCHeaderOutputPathForAtMostOnePrimary(),
+            Instance.getMainModule(), opts.ImplicitObjCHeaderPath,
+            moduleIsPublic);
```

Comment:
I don't see how to.
Without this fix, the test causes the compiler to falsify an assertion. If you ask for a -typecheck in batch mode, it calls `getObjCHeaderOutputPathForAtMostOnePrimary`, but that trips an assertion because there is >1 primary, even though no objC header has been requested.
The test *could* be folded into `getObjCHeaderOutputPathForAtMostOnePrimary` but that would be inconsistent with the other `ForAtMostOnePrimary` functions.

Reply:
What changed that makes this come up now, though?

Reply:
Good question. Will investigate after lunch.

Reply:
I believe this to be a bug that has been there for a while. I ran an older version of the compiler with the following arguments: `-frontend -typecheck -primary-file /s/io/1.swift -primary-file /s/io/2.swift` (where each .swift file is empty) and it falsified an assertion.
It is the combination of `-typecheck` with > 1 primary that excites the bug.

Reply:
Got it. I guess we can sneak it in here then.